### ISSUE TITLE
add some description on WHAM?

### ISSUE THREAD

**Issue by smutao** [alchemistry/alchemical-best-practices#123]
In section 8.3, many estimators were summarized. Is it possible to also have some description on WHAM? Because WHAM was mentioned a few times in the paper (especially in the "recommendations"). Thanks a lot for contributing this great review article.

**Comment by jchodera** [comment]
This is a good suggestion! In general, we now prefer [MBAR](https://doi.org/10.1063/1.2978177) and related statistically optimal binless formulations to histogram-based WHAM methods because they are computationally and statistically superior, but the relationships here should be clearly explain with all the relevant references---and summaries of how they are useful in illuminating properties---should be provided.

**Comment by ppxasjsm** [comment]
I agree with John's statement here. Actually, the TRAM papers do lay out the connections quite well, so this is something I can probably tackle too.

**Comment by agrossfield** [comment]
I largely agree. Just about the only upside to wham for this kind of application is that it’s faster and less memory intensive because your storage scales as the histogram size rather than the number of data points, but there are very few if any cases where the difference is enough to actually matter. Given that wham never really caught on in the alchemical space, I’m not sure it’s worth more than a casual mention.

Cheers,

Alan


________________________________
Dr. Alan Grossfield
Dept of Biochemistry and Biophysics
University of Rochester Medical Center
Phone: 585 276 4193
http://membrane.urmc.rochester.edu
https://orcid.org/0000-0002-5877-2789
Pronouns: He/his



From: John Chodera ***@***.***>
Reply-To: alchemistry/alchemical-best-practices ***@***.***>
Date: Sunday, February 13, 2022 at 11:21 PM
To: alchemistry/alchemical-best-practices ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [EXT] Re: [alchemistry/alchemical-best-practices] add some description on WHAM? (Issue #123)


This is a good suggestion! In general, we now prefer MBAR<https://urldefense.proofpoint.com/v2/url?u=https-3A__doi.org_10.1063_1.2978177&d=DwMCaQ&c=4sF48jRmVAe_CH-k9mXYXEGfSnM3bY53YSKuLUQRxhA&r=49qnaP-kgQR_zujl5kbj_PmvQeXyz1NAoiLoIzsc27zuRX32UDM2oX8NQCaAsZzH&m=EAv0t7t8SMc_XK_6QGRmEi4HLiaQBP5KjdVr_dAChAA&s=onFKK_SPX8ZHgZ_cV-lE5Xy2zv0i-AkKParaFvo5TyQ&e=> and related statistically optimal binless formulations to histogram-based WHAM methods because they are computationally and statistically superior, but the relationships here should be clearly explain with all the relevant references---and summaries of how they are useful in illuminating properties---should be provided.

—
Reply to this email directly, view it on GitHub<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_alchemistry_alchemical-2Dbest-2Dpractices_issues_123-23issuecomment-2D1038624368&d=DwMCaQ&c=4sF48jRmVAe_CH-k9mXYXEGfSnM3bY53YSKuLUQRxhA&r=49qnaP-kgQR_zujl5kbj_PmvQeXyz1NAoiLoIzsc27zuRX32UDM2oX8NQCaAsZzH&m=EAv0t7t8SMc_XK_6QGRmEi4HLiaQBP5KjdVr_dAChAA&s=CTfbnoqBRPeZHJUsP4SogmCAt8BDnL_BoRVXvklvBAI&e=>, or unsubscribe<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_ADH754QXXQUEWOHISJ6XV73U3B7ODANCNFSM5ODATUBQ&d=DwMCaQ&c=4sF48jRmVAe_CH-k9mXYXEGfSnM3bY53YSKuLUQRxhA&r=49qnaP-kgQR_zujl5kbj_PmvQeXyz1NAoiLoIzsc27zuRX32UDM2oX8NQCaAsZzH&m=EAv0t7t8SMc_XK_6QGRmEi4HLiaQBP5KjdVr_dAChAA&s=9JlXU7VYXUHDFM59RIW7gyqm4K0RxXZLvg4xzFjZXa0&e=>.
Triage notifications on the go with GitHub Mobile for iOS<https://urldefense.proofpoint.com/v2/url?u=https-3A__apps.apple.com_app_apple-2Dstore_id1477376905-3Fct-3Dnotification-2Demail-26mt-3D8-26pt-3D524675&d=DwMCaQ&c=4sF48jRmVAe_CH-k9mXYXEGfSnM3bY53YSKuLUQRxhA&r=49qnaP-kgQR_zujl5kbj_PmvQeXyz1NAoiLoIzsc27zuRX32UDM2oX8NQCaAsZzH&m=EAv0t7t8SMc_XK_6QGRmEi4HLiaQBP5KjdVr_dAChAA&s=AvgzSIrYq3rNEBvOdLjttOS7aW-HJfKGNA82MC7HasA&e=> or Android<https://urldefense.proofpoint.com/v2/url?u=https-3A__play.google.com_store_apps_details-3Fid-3Dcom.github.android-26referrer-3Dutm-5Fcampaign-253Dnotification-2Demail-2526utm-5Fmedium-253Demail-2526utm-5Fsource-253Dgithub&d=DwMCaQ&c=4sF48jRmVAe_CH-k9mXYXEGfSnM3bY53YSKuLUQRxhA&r=49qnaP-kgQR_zujl5kbj_PmvQeXyz1NAoiLoIzsc27zuRX32UDM2oX8NQCaAsZzH&m=EAv0t7t8SMc_XK_6QGRmEi4HLiaQBP5KjdVr_dAChAA&s=4ETWwS0BRHqeTwDmCfxE_rTXNASlCiNviOSTA90CJ9E&e=>.

**Comment by jchodera** [comment]
> I largely agree. Just about the only upside to wham for this kind of application is that it’s faster and less memory intensive because your storage scales as the histogram size rather than the number of data points, but there are very few if any cases where the difference is enough to actually matter. Given that wham never really caught on in the alchemical space, I’m not sure it’s worth more than a casual mention.

@agrossfield : The reason it never caught on is that in cases where the alchemical coordinate is not simply linearly scaling potentials requires highly multidimensional histograms where the efficiency gain is lost. In parallel tempering, histogram WHAM is absolutely the winner, and can even be used to initialize histogramless WHAM / MBAR which converges extremely quickly.

@ppxasjsm : Great point! TRAM should even replace MBAR as the optimal analysis method in cases where the simulation is not thoroughly sampled at equilibrium (which is basically all the time).

**Comment by agrossfield** [comment]
﻿I agree. My point was simply that there isn’t a community of folks using wham (or a bunch of literature) that you need to counter as being non-best practices. From what I’ve seen, 99% of papers doing alchemy either do perturbation or mbar (or thermodynamic integration, which is a separate issue), and so you just need to recommend doing mbar or tram (I’ll defer to your superior expertise as to which makes sense).

________________________________
Dr. Alan Grossfield
Dept of Biochemistry and Biophysics
University of Rochester Medical Center
Phone: 585 276 4193
http://membrane.urmc.rochester.edu
https://orcid.org/0000-0002-5877-2789
Pronouns: He/his



From: John Chodera ***@***.***>
Reply-To: alchemistry/alchemical-best-practices ***@***.***>
Date: Monday, February 14, 2022 at 11:18 AM
To: alchemistry/alchemical-best-practices ***@***.***>
Cc: "Grossfield, Alan" ***@***.***>, Mention ***@***.***>
Subject: [EXT] Re: [alchemistry/alchemical-best-practices] add some description on WHAM? (Issue #123)


I largely agree. Just about the only upside to wham for this kind of application is that it’s faster and less memory intensive because your storage scales as the histogram size rather than the number of data points, but there are very few if any cases where the difference is enough to actually matter. Given that wham never really caught on in the alchemical space, I’m not sure it’s worth more than a casual mention.

@agrossfield<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_agrossfield&d=DwMFaQ&c=4sF48jRmVAe_CH-k9mXYXEGfSnM3bY53YSKuLUQRxhA&r=49qnaP-kgQR_zujl5kbj_PmvQeXyz1NAoiLoIzsc27zuRX32UDM2oX8NQCaAsZzH&m=D6rOAYHxn5D6xA1pbpfeT5KM1DnFRuaPMM83gZucqfI&s=XTB6a32VPbCgTt0DEDsS-7m2DRSWkNNoIuxPEi7D4TE&e=> : The reason it never caught on is that in cases where the alchemical coordinate is not simply linearly scaling potentials requires highly multidimensional histograms where the efficiency gain is lost. In parallel tempering, histogram WHAM is absolutely the winner, and can even be used to initialize histogramless WHAM / MBAR which converges extremely quickly.

@ppxasjsm<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_ppxasjsm&d=DwMFaQ&c=4sF48jRmVAe_CH-k9mXYXEGfSnM3bY53YSKuLUQRxhA&r=49qnaP-kgQR_zujl5kbj_PmvQeXyz1NAoiLoIzsc27zuRX32UDM2oX8NQCaAsZzH&m=D6rOAYHxn5D6xA1pbpfeT5KM1DnFRuaPMM83gZucqfI&s=2LwqXSIBzgU_LQaDhEusre67QEkf5Jagc7X2xnOkq7Y&e=> : Great point! TRAM should even replace MBAR as the optimal analysis method in cases where the simulation is not thoroughly sampled at equilibrium (which is basically all the time).

—
Reply to this email directly, view it on GitHub<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_alchemistry_alchemical-2Dbest-2Dpractices_issues_123-23issuecomment-2D1039232679&d=DwMFaQ&c=4sF48jRmVAe_CH-k9mXYXEGfSnM3bY53YSKuLUQRxhA&r=49qnaP-kgQR_zujl5kbj_PmvQeXyz1NAoiLoIzsc27zuRX32UDM2oX8NQCaAsZzH&m=D6rOAYHxn5D6xA1pbpfeT5KM1DnFRuaPMM83gZucqfI&s=fvbCTFa8JpxmwkUUnmZdt4pVG5ayurpxpf9-tV4YdjU&e=>, or unsubscribe<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_ADH754XXMO4PMQZ35AYVXR3U3EOS3ANCNFSM5ODATUBQ&d=DwMFaQ&c=4sF48jRmVAe_CH-k9mXYXEGfSnM3bY53YSKuLUQRxhA&r=49qnaP-kgQR_zujl5kbj_PmvQeXyz1NAoiLoIzsc27zuRX32UDM2oX8NQCaAsZzH&m=D6rOAYHxn5D6xA1pbpfeT5KM1DnFRuaPMM83gZucqfI&s=qOmB3DRC62QKLI_Ts2BNkjOvlYM5UkEsZObNTAE2R1A&e=>.
Triage notifications on the go with GitHub Mobile for iOS<https://urldefense.proofpoint.com/v2/url?u=https-3A__apps.apple.com_app_apple-2Dstore_id1477376905-3Fct-3Dnotification-2Demail-26mt-3D8-26pt-3D524675&d=DwMFaQ&c=4sF48jRmVAe_CH-k9mXYXEGfSnM3bY53YSKuLUQRxhA&r=49qnaP-kgQR_zujl5kbj_PmvQeXyz1NAoiLoIzsc27zuRX32UDM2oX8NQCaAsZzH&m=D6rOAYHxn5D6xA1pbpfeT5KM1DnFRuaPMM83gZucqfI&s=5hNQyAfqqqX8sy28h5mqIDXzuJf6FfyulfXWPwfTdto&e=> or Android<https://urldefense.proofpoint.com/v2/url?u=https-3A__play.google.com_store_apps_details-3Fid-3Dcom.github.android-26referrer-3Dutm-5Fcampaign-253Dnotification-2Demail-2526utm-5Fmedium-253Demail-2526utm-5Fsource-253Dgithub&d=DwMFaQ&c=4sF48jRmVAe_CH-k9mXYXEGfSnM3bY53YSKuLUQRxhA&r=49qnaP-kgQR_zujl5kbj_PmvQeXyz1NAoiLoIzsc27zuRX32UDM2oX8NQCaAsZzH&m=D6rOAYHxn5D6xA1pbpfeT5KM1DnFRuaPMM83gZucqfI&s=od2i4OkjJVYcxMQAZU0WkO-igc7xNrHuAwZPTrr_1gs&e=>.

**Comment by smutao** [comment]
Hello Profs. Chodera, Mey and Grossfield, 
thank you for considering my suggestions. 
YT

**Comment by ppxasjsm** [comment]
Will leave this open until it has actually been addressed in the text!

**Comment by fjclark** [comment]
@ppxasjsm or @jchodera, are either of you happy to address this (before V2.0 is finalised at the end of next May)? Thanks.
